storage/client/src/main/java/com/azure/storage/queue/QueueAsyncClient.java

```diff
@@ -1,28 +1,276 @@
+        return client.messages().peekWithRestResponseAsync(queueName, maxMessages, null, null, Context.NONE)
+            .flatMapMany(response -> Flux.fromIterable(response.value()));
+    }
+
+    /**
+     * Updates the message in the queue
+     * @param messageText Updated value for the message
+     * @param messageId Id of the message
+     * @param popReceipt Unique identifier that must match the message for it to be updated
+     * @param visibilityTimeout How long the message will be invisible in the queue in seconds
+     * @return the updated message information
+     */
+    public Mono<Response<UpdatedMessage>> updateMessage(String messageText, String messageId, String popReceipt, Duration visibilityTimeout) {
+        QueueMessage message = new QueueMessage().messageText(messageText);
+        return client.messageIds().updateWithRestResponseAsync(queueName, messageId, message, popReceipt, (int) visibilityTimeout.getSeconds(), Context.NONE)
+            .map(this::getUpdatedMessageResponse);
+    }
+
+    /**
+     * Deletes the message from the queue
+     * @param messageId Id of the message
+     * @param popReceipt Unique identifier that must match the message for it to be deleted
+     * @return an empty response
+     */
+    public Mono<VoidResponse> deleteMessage(String messageId, String popReceipt) {
+        return client.messageIds().deleteWithRestResponseAsync(queueName, messageId, popReceipt, Context.NONE)
+            .map(VoidResponse::new);
+    }
+
+    /*
+     * Maps the HTTP headers returned from the service to the expected response type
+     * @param response Service response
+     * @return Mapped response
+     */
+    private Response<QueueProperties> getQueuePropertiesResponse(QueuesGetPropertiesResponse response) {
+        QueueGetPropertiesHeaders propertiesHeaders = response.deserializedHeaders();
+        QueueProperties properties = new QueueProperties(propertiesHeaders.metadata(), propertiesHeaders.approximateMessagesCount());
+        return mapResponse(response, properties);
+    }
+
+    /*
+     * Maps the HTTP headers returned from the service to the expected response type
+     * @param response Service response
+     * @return Mapped response
+     */
+    private Response<UpdatedMessage> getUpdatedMessageResponse(MessageIdsUpdateResponse response) {
+        MessageIdUpdateHeaders headers = response.deserializedHeaders();
+        UpdatedMessage updatedMessage = new UpdatedMessage(headers.popReceipt(), headers.timeNextVisible());
+        return mapResponse(response, updatedMessage);
+    }
+
+    /*
+     * Maps a response from the service to the expected response for the client
+     * @param response Service response
+     * @param value Service response value
+     * @param <T> Type of teh service response value
+     * @return Mapped response
+     */
+    private <T> SimpleResponse<T> mapResponse(Response response, T value) {
```

Comment:
Can we put this method to Azure core SimpleResponse. It is frequently used in different storage clients.